storage/src/main/java/org/apache/kafka/server/log/remote/storage/RemoteLogManager.java

```diff
@@ -1031,6 +1031,7 @@ private void copyLogSegment(UnifiedLog log, LogSegment segment, RemoteLogSegment
                 customMetadata = remoteStorageManagerPlugin.get().copyLogSegmentData(copySegmentStartedRlsm, segmentData);
             } catch (RemoteStorageException e) {
                 logger.info("Copy failed, cleaning segment {}", copySegmentStartedRlsm.remoteLogSegmentId());
+                recordLagStats(log);
```

Comment:
This patch handles the error when interacting with the RemoteStorageManager. Should we also handle any error when interacting with RemoteLogMetadataManager? 

The segmentsLag / bytesLag metric should show the real-time value whenever there is any error in interacting with RSM or RLMM plugin.

Reply:
Yes, we should also handle the RLMM-related methods. The following methods should be covered:
- `addRemoteLogSegmentMetadata`
- `updateRemoteLogSegmentMetadata`

Additionally, there's another case at line 579:
```
remoteStorageManagerPlugin.get().deleteLogSegmentData(segmentMetadata);
```
Should we also consider adding more explicit handling here?
```
try {
    remoteStorageManagerPlugin.get().deleteLogSegmentData(segmentMetadata);
} catch (RemoteStorageException e) {
    brokerTopicStats.topicStats(topic).failedRemoteDeleteRequestRate().mark();
    brokerTopicStats.allTopicsStats().failedRemoteDeleteRequestRate().mark();
    throw e;
}
```

Reply:
I think the `recordLagStats(log)` can be moved to either catch / finally block of `copyLogSegmentsToRemote` to handle all the cases.